bucket4j/src/main/java/com/linecorp/armeria/server/throttling/tokenbucket/BandwidthLimit.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+    /**
+     * Creates new {@link BandwidthLimit}. Specifies limitation in
+     * <a href="https://github.com/vladimir-bukhtoyarov/bucket4j/blob/1.3/doc-pages/token-bucket-brief-overview.md#token-bucket-algorithm">classic
+     * interpretation</a> of token-bucket algorithm.
+     *
+     * @param limit the bucket size - defines the count of tokens which can be held by bucket
+     *              and defines the speed at which tokens are regenerated in bucket
+     * @param overdraftLimit defines the maximum overdraft count of tokens which can be held by bucket,
+     *                       this must exceed the {@code limit}
+     * @param period the time window, during which the tokens will be regenerated
+     */
+    public static BandwidthLimit of(long limit, long overdraftLimit, Duration period) {
+        return new BandwidthLimit(limit, overdraftLimit, 0L, period);
+    }
+
+    /**
+     * Creates new {@link BandwidthLimit}.
+     * Specifies easy limitation of {@code limit} tokens per {@code period} time window.
+     * @param limit the bucket size - defines the maximum count of tokens which can be held by bucket
+     *              and defines the speed at which tokens are regenerated in bucket
+     * @param period the time window, during which the tokens will be regenerated
+     */
+    public static BandwidthLimit of(long limit, Duration period) {
+        return new BandwidthLimit(limit, 0L, 0L, period);
+    }
+
+    /**
+     * Creates a new {@link BandwidthLimit} that computes {@code limit}, {@code overdraftLimit},
+     * {@code initialSize} and {@code period} from a semicolon-separated {@code specification} string
+     * that conforms to the following format,
+     * as per <a href="https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html">RateLimit Header Scheme for HTTP</a>:
+     * <pre>{@code
+     * <limit>;window=<period(in seconds)>[;burst=<overdraftLimit>][;initial=<initialSize>]
+     * }</pre>
+     * All {@code specification} string elements must come in the defined order.
+     * For example:
+     * <ul>
+     *   <li>{@code 100;window=60;burst=1000} ({@code limit}=100, {@code overdraftLimit}=1000,
+     *       {@code initialSize} and {@code period}=60seconds)</li>
+     *   <li>{@code 100;window=60;burst=1000;initial=20} ({@code limit}=100, {@code overdraftLimit}=1000,
+     *       {@code initialSize}=20 and {@code period}=60seconds)</li>
+     *   <li>{@code 5000;window=1} ({@code limit}=5000 and {@code period}=1second)</li>
+     * </ul>
+     *
+     * @param specification the specification used to create a {@link BandwidthLimit}
+     */
+    public static BandwidthLimit of(String specification) {
+        return TokenBucketSpec.parseBandwidthLimit(specification);
+    }
+
+    /**
+     * The bucket size - defines the count of tokens which can be held by bucket
+     * and defines the speed at which tokens are regenerated in bucket.
+     * @return Bucket size.
```

Comment:
- `Returns the bucket size, which defines ...`
- Probably no need to specify `@return` tag because summary implies it.